Randomise beatmap playback order on startup

peppy opened the pull request:
Closes #6135.

Note that due to limitations of `BindableList`, newly imported beatmaps are always added at the end. This doesn't seem like much of an issue, though.

peppy committed:
Randomise beatmap playback order on startup

Closes #6135.

smoogipoo commented on osu.Game/Overlays/MusicController.cs:
```diff
@@ -24,7 +25,7 @@ public class MusicController : Component, IKeyBindingHandler<GlobalAction>
         private BeatmapManager beatmaps { get; set; }
 
-        private List<BeatmapSetInfo> beatmapSets;
+        public readonly BindableList<BeatmapSetInfo> BeatmapSets = new BindableList<BeatmapSetInfo>();
```
This should at the very least be an IBindableList if this is how this is going to be done.

peppy committed:
Merge remote-tracking branch 'upstream/master' into randomise-song-playback

peppy committed:
Expose as IBindableList

peppy committed:
Localise OrderChanged handling and fix callbacks

The dragged item's position now only updates after the drag finishes. Local handling changes were required to ignore the bindable remove/add events that are fired as a result.

smoogipoo committed:
Merge remote-tracking branch 'Joehuu/master' into randomise-song-playback

smoogipoo committed:
Use resolved attribute for music controller

smoogipoo committed:
Cleanup draggedItem usages and make them more safe
